components/proxy/src/main/java/com/hotels/styx/metrics/reporting/sets/NettyAllocatorMetricSet.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2013-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.metrics.reporting.sets;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import io.netty.buffer.PooledByteBufAllocatorMetric;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * Creates a set of gauges that monitor metrics of netty {#PooledByteBufAllocatorMetric} instance.
+ */
+public class NettyAllocatorMetricSet implements MetricSet {
+
+    private final PooledByteBufAllocatorMetric metric;
+
+    public NettyAllocatorMetricSet(PooledByteBufAllocatorMetric metric) {
+        this.metric = metric;
+    }
+
+    @Override
+    public Map<String, Metric> getMetrics() {
+        Map<String, Metric> gauges = new HashMap<>();
+        gauges.put("pooled-allocator.numDirectArenas", (Gauge<Integer>) metric::numDirectArenas);
+        gauges.put("pooled-allocator.numHeapArenas", (Gauge<Integer>) metric::numHeapArenas);
+        gauges.put("pooled-allocator.numTreadLocalCaches", (Gauge<Integer>) metric::numThreadLocalCaches);
+        gauges.put("pooled-allocator.tinyCacheSize", (Gauge<Integer>) metric::tinyCacheSize);
+        gauges.put("pooled-allocator.smallCacheSize", (Gauge<Integer>) metric::smallCacheSize);
+        gauges.put("pooled-allocator.normalCacheSize", (Gauge<Integer>) metric::normalCacheSize);
+        gauges.put("pooled-allocator.chunkSize", (Gauge<Integer>) metric::chunkSize);
+        gauges.put("pooled-allocator.usedDirectMemory", (Gauge<Long>) metric::usedDirectMemory);
+        gauges.put("pooled-allocator.usedHeapMemory", (Gauge<Long>) metric::usedHeapMemory);
+        return unmodifiableMap(gauges);
```

Comment:
`Collections.unmodifiableMap` keeps a reference to the original map.
Since the backing map is not going to change we can just using guava's `ImmutableMap.copyOf`.